navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/instruction/DefaultInstructionEngine.java

```diff
@@ -0,0 +1,223 @@
+package com.mapbox.services.android.navigation.v5.instruction;
+
+import android.text.TextUtils;
+import android.util.SparseArray;
+
+import com.mapbox.services.android.navigation.v5.NavigationConstants;
+import com.mapbox.services.android.navigation.v5.RouteProgress;
+import com.mapbox.services.api.directions.v5.models.StepManeuver;
+import com.mapbox.services.api.utils.turf.TurfConstants;
+import com.mapbox.services.api.utils.turf.TurfHelpers;
+
+import java.text.DecimalFormat;
+import java.util.Locale;
+
+class DefaultInstructionEngine extends SparseArray<DefaultInstructionEngine.InstructionBuilder> {
```

Comment:
Even though the class is only visible inside the package, it might be good to add some basic Javadoc here including the `@since 0.4.0` tag.